core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
```

Comment:
so I took a quick look at this and I agree with you that ideally its returning ones actually removed, but right now I don't see any of the backends actually returning anything other then true unless its something catastrophic where it wouldn't matter if the executor list is 1 or many.
 Yarn mode right now always returns true (ApplicationMaster KillExecutors endpoint). Standalone mode only return false if the app doesn't exist so if we pass 1 or many executors doesn't matter, mesos mode only returns false if the mesosdriver is null.

So changing the interface to do the right things seems like it can be a separate issue then this one.  or perhaps I missed something?


But it looks like there are a few cases that mesos and standalone mode do return false.